Symfony/CS/Tests/Fixer/Contrib/PreIncrementFixerTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests\Fixer\Contrib;
+
+use Symfony\CS\Tests\Fixer\AbstractFixerTestBase;
+
+/**
+ * @author Gregor Harlan <gharlan@web.de>
+ */
+class PreIncrementFixerTest extends AbstractFixerTestBase
+{
+    /**
+     * @dataProvider provideExamples
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->makeTest($expected, $input);
+    }
+
+    public function provideExamples()
+    {
+        return array(
+            array(
+                '<?php ++$a;',
+                '<?php $a++;',
+            ),
+            array(
+                '<?php ++$$a;',
+                '<?php $$a++;',
+            ),
+            array(
+                '<?php ++${"a"};',
+                '<?php ${"a"}++;',
+            ),
+            array(
+                '<?php --$a;',
+                '<?php $a--;',
+            ),
+            array(
+                '<?php foo(); ++$a;',
+                '<?php foo(); $a++;',
+            ),
+            array(
+                '<?php if (true) { ++$a; }',
+                '<?php if (true) { $a++; }',
+            ),
+            array(
+                '<?php if (true) {} ++$a;',
+                '<?php if (true) {} $a++;',
+            ),
+            array(
+                '<?php for ($i = 0; $i < $count; ++$i) {}',
+                '<?php for ($i = 0; $i < $count; $i++) {}',
+            ),
+
+            array('<?php $a = $b++;'),
+            array('<?php $a + $b++;'),
+            array('<?php $a++ + $b;'),
+            array('<?php foo($b++);'),
+            array('<?php foo($a, $b++);'),
+            array('<?php $a[$b++];'),
+            array('<?php echo $a++;'),
+            array('<?php $a->foo++;'),
```

Comment:
And what about the next 5 lines? Also see discussion in https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/831#issuecomment-87317472
I want to be sure which cases should be converted before I implement them, because they are more difficult to handle.

Extreme example:

```php
$a->{$b."foo"}->bar[$c]->$baz++;
```